Lib/fontTools/varLib/__init__.py

```diff
@@ -265,9 +269,10 @@ def build(designspace_filename, master_finder=lambda s:s, axisMap=None):
 	"""
 
 	ds = designspace.load(designspace_filename)
-	axes = ds['axes']
-	masters = ds['masters']
-	instances = ds['instances']
+	axes = ds['axes'] if 'axes' in ds else []
+	if 'sources' not in ds or not ds['sources']:
+		raise VarLibError("no 'sources' defined in .designspace")
+	instances = ds['instances'] if 'instances' in ds else []
```

Comment:
BTW, I'm putting together some test cases that will exercise `fontTools.varLib.build()` and `fontTools.varLib.interpolate_layout.interpolate_layout()`. I think these will help a lot validating this and upcoming changes.